base/cvd/cuttlefish/common/libs/fs/shared_fd.cpp

```diff
@@ -524,6 +527,24 @@ SharedFD SharedFD::Mkstemp(std::string* path) {
   }
 }
 
+Result<std::pair<SharedFD, std::string>> SharedFD::Mkostemp(
```

Comment:
https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs writes

> The output of a C++ function is naturally provided via a return value and sometimes via output parameters (or in/out parameters).
>
> Prefer using return values over output parameters: they improve readability, and often provide the same or better performance.

Is matching the type signature of `stdlib.h` more important than following this part of the style guide?

Reply:
I also prefer the return value and taking care of the `path` copy inside `Mkostemp` instead of leaving it to the caller, enforcing the caller has both values.

Reply:
In my opinion, yes. Especially because it doesn't forbid the use of in/out parameters, it just says "prefer" the return approach.
In any case I don't have a strong opinion here, that's what I marked it as a "nit", either way is fine by me.

Reply:
> taking care of the path copy inside Mkostemp

No copy would be needed with the in/out parameter approach. I also don't see the value of having both the template and the path after the file has been created, although it doesn't hurt either.